src/ukify/ukify.py

```diff
@@ -679,10 +679,12 @@ def make_uki(opts):
 
     call_systemd_measure(uki, linux, opts=opts)
 
-    # UKI creation
+    # UKI or addon creation - addons don't use the stub so we add SBAT manually
 
     if linux is not None:
         uki.add_section(Section.create('.linux', linux, measure=True))
+    elif opts.sbat:
+        uki.add_section(Section.create('.sbat', opts.sbat, measure=False))
```

Comment:
I think this creates a second `.sbat` section. Have you tested this works? I believe shim looks at all sbat sections, but I might be wrong here...

Reply:
ie: UKIs get their section from the stub, but there's no stub in an addon

Reply:
> ie: UKIs get their section from the stub, but there's no stub in an addon

Huh? ukify still uses the stub image to build the addons, no? So the sbat section from it is still in the PE image.

The ukify code only checks for duplicate sections about to be added, but not against the sections already in the stub PE image. So adding a sbat section should produce a second sbat section. You wanna look at `objdump -h` on an addon.

Reply:
No, the stub is not used. Well, it's not enforced-off I suppose, so I'll add that, but it is not intended to be used.

```
objdump -h /tmp/img/efi/loader/addons/test.addon.efi

/tmp/img/efi/loader/addons/test.addon.efi:     file format pei-x86-64

Sections:
Idx Name          Size      VMA               LMA               File off  Algn
  0 .text         00007aa0  0000000000004000  0000000000004000  00000400  2**4
                  CONTENTS, ALLOC, LOAD, READONLY, CODE
  1 .reloc        0000000a  000000000000c000  000000000000c000  00008000  2**0
                  CONTENTS, ALLOC, LOAD, READONLY, DATA
  2 .data         00002168  000000000000d000  000000000000d000  00008200  2**5
                  CONTENTS, ALLOC, LOAD, DATA
  3 .dynamic      00000110  0000000000010000  0000000000010000  0000a400  2**3
                  CONTENTS, ALLOC, LOAD, DATA
  4 .rela         00000e58  0000000000011000  0000000000011000  0000a600  2**3
                  CONTENTS, ALLOC, LOAD, READONLY, DATA
  5 .dynsym       00000378  0000000000012000  0000000000012000  0000b600  2**3
                  CONTENTS, ALLOC, LOAD, READONLY, DATA
  6 .cmdline      00000003  0000000000013000  0000000000013000  0000ba00  2**2
                  CONTENTS, ALLOC, LOAD, READONLY, DATA
  7 .uname        00000013  0000000000014000  0000000000014000  0000bc00  2**2
                  CONTENTS, ALLOC, LOAD, READONLY, DATA
  8 .sbat         000000c1  0000000000015000  0000000000015000  0000be00  2**2
                  CONTENTS, ALLOC, LOAD, READONLY, DATA
```

Reply:
> No, the stub is not used. Well, it's not enforced-off I suppose, so I'll add that, but it is not intended to be used.

I am very confused. ukify requires a PE image/stub for operation as a template. It doesn't generate a PE from scratch. What are you using as a addon/stub template?

From the looks of it, it's the default picked up from /usr of the mkosi image? If that one is supposed to use the compiled one from git, I am confused, because it should not have a rela/dynsym/dynamic section.

Reply:
yeah you are right, I was just looking for the .sdmagic section